.github/workflows/maven.yml

```diff
@@ -14,7 +14,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        maven: [3.3.1, 3.5.4, 3.6.3, 3.8.1]
+        maven: [3.3.1, 3.5.4, 3.6.3, 3.8.7, 3.9.0]
```

Comment:
you can probably get rid of 3.3.1 and 3.5.4 as it's really supported but the Apache Maven ;)

Reply:
Is there a documented policy for Maven versions supported by Apache?

The _de facto_ policy for this library has been to support as wide a range as possible, to avoid the need to maintain multiple branches for backward compatibility. I personally wouldn't object to dropping older versions (people can always use older versions of this library, especially since it isn't changing much) but I think there ought to be a reason to do so. Supporting these old versions is not really adding maintenance burden since the testing is automated and run in parallel. I also think that doing so should require a major version increase.

If anything, I'd want to drop support for anything but 3.9.0 if we were to do a new major version, as it would allow us to remove the workaround for [MNG-6965](https://issues.apache.org/jira/browse/MNG-6965) that was fixed in 3.9.0 (discussed previously in #75).

Reply:
uhm good point I cannot see real documented policy, I will search and if none probably add one.
As an indication all the plugins maintained by Apache Maven folks have been migrated to at least 3.6.3.

Reply:
Supporting the current version and two previous minor versions seems like a sensible policy. It seems [there was never a 3.7.x series](https://maven.apache.org/docs/3.8.1/release-notes.html#why-does-this-version-have-the-value-3-8-1). That said, I still wouldn't want to drop support for older versions as part of this specific change. It should be discussed as an independent issue.

Reply:
no worries. sounds good. adding test of 4.0.0-alpha would be good as well :)

Reply:
Yes, good suggestion.